src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -396,8 +396,15 @@ private async Task<CreationResultStatus> EnterTemplateManipulationFlowAsync()
             }
 
             TemplateResolutionResult templateResolutionResult = TemplateResolver.GetTemplateResolutionResult(_settingsLoader.UserTemplateCache.TemplateInfo, _hostDataLoader, _commandInput, _defaultLanguage);
-            TemplateInvocationAndAcquisitionCoordinator invocationCoordinator = new TemplateInvocationAndAcquisitionCoordinator(_settingsLoader, _commandInput, _templateCreator, _hostDataLoader, _telemetryLogger, _defaultLanguage, CommandName, _inputGetter, _callbacks);
-            return await invocationCoordinator.CoordinateInvocationOrAcquisitionAsync();
+            if (templateResolutionResult.TryGetSingularInvokableMatch(out ITemplateMatchInfo templateToInvoke, out TemplateResolutionResult.Status resultStatus))
+            {
+                TemplateInvocationAndAcquisitionCoordinator invocationCoordinator = new TemplateInvocationAndAcquisitionCoordinator(_settingsLoader, _commandInput, _telemetryLogger, CommandName, _inputGetter, _callbacks);
+                return await invocationCoordinator.CoordinateInvocationOrAcquisitionAsync(templateToInvoke);
```

Comment:
please configure the awaitable to false.

Reply:
In theory dropping `async/await` and returning `Task` directly would be even nicer... Ofc other 2 returns will need `Task.FromResult()`